### ISSUE TITLE
./install.sh does not work on CentOS 7 (be stuck) 

### ISSUE THREAD

**Issue by enjoy2303** [getsentry/self-hosted#2133]
### Self-Hosted Version

22.10

### CPU Architecture

x86_64

### Docker Version

23.0.5

### Docker Compose Version

2.6.0

### Steps to Reproduce

./install.sh

### Expected Result

Installation finished

### Actual Result

▶ Parsing command line ...

▶ Initializing Docker Compose ...

▶ Setting up error handling ...

▶ Checking for latest commit ... 
skipped

▶ Checking minimum requirements ...

▶ Turning things off ...
 Image snuba-cleanup-self-hosted-local:latest  Removing
 Image sentry-cleanup-self-hosted-local:latest  Removing
 Image sentry-cleanup-self-hosted-local:latest  Removed
 Image snuba-cleanup-self-hosted-local:latest  Removed

▶ Creating volumes for persistent storage ...
Created sentry-clickhouse.
Created sentry-data.
Created sentry-kafka.
Created sentry-postgres.
Created sentry-redis.
Created sentry-symbolicator.
Created sentry-zookeeper.

▶ Ensuring files from examples ...
../sentry/sentry.conf.py already exists, skipped creation.
../sentry/config.yml already exists, skipped creation.
../symbolicator/config.yml already exists, skipped creation.
../sentry/requirements.txt already exists, skipped creation.

▶ Generating secret key ...

▶ Replacing TSDB ...

▶ Fetching and updating Docker images ...
22.1.0: Pulling from getsentry/sentry
Digest: sha256:6d1aa6de79eccb6f1355e5da545a2f7533b632b8548a870d3335e83696bec8be
Status: Image is up to date for getsentry/sentry:22.1.0
docker.io/getsentry/sentry:22.1.0

▶ Building and tagging Docker images ...

#1 [internal] load build definition from Dockerfile
#1 transferring dockerfile: 1.09kB 0.0s done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 transferring context: 2B 0.0s done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/getsentry/snuba:22.1.0
#3 DONE 0.0s

#4 [1/6] FROM docker.io/getsentry/snuba:22.1.0
#4 DONE 0.0s

#5 [2/6] RUN sed -i 's#http://deb.debian.org#https://mirrors.163.com#g' /etc/apt/sources.list
#5 CACHED

#6 [internal] load build context
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 1.09kB done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/getsentry/sentry:22.1.0
#3 DONE 0.0s

#4 [internal] load build context
#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 1.09kB done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/getsentry/symbolicator:0.4.1
#3 DONE 0.0s

#4 [1/6] FROM docker.io/getsentry/symbolicator:0.4.1
#4 DONE 0.0s

#5 [2/6] RUN sed -i 's#http://deb.debian.org#https://mirrors.163.com#g' /etc/apt/sources.list
#5 CACHED

#6 [internal] load build context
#6 transferring context: 95B done
#6 DONE 0.0s

#7 [3/6] RUN sed -i 's|#*deb http://security.debian.org|deb https://mirrors.cqu.edu.cn|' /etc/apt/sources.list /etc/apt/sources.list
#6 transferring context: 95B 0.0s done
#6 DONE 0.0s

#7 [3/6] RUN sed -i 's|#*deb http://security.debian.org|deb https://mirrors.cqu.edu.cn|' /etc/apt/sources.list /etc/apt/sources.list
#4 transferring context: 95B done
#4 DONE 0.0s

#5 [1/6] FROM docker.io/getsentry/sentry:22.1.0
#5 DONE 0.0s

#6 [2/6] RUN sed -i 's#http://deb.debian.org#https://mirrors.163.com#g' /etc/apt/sources.list
#6 CACHED

#7 [3/6] RUN sed -i 's|#*deb http://security.debian.org|deb https://mirrors.cqu.edu.cn|' /etc/apt/sources.list /etc/apt/sources.list
#1 [1/6] FROM docker.io/getsentry/snuba:22.1.0
#1 DONE 0.0s

#2 [2/6] RUN sed -i 's#http://deb.debian.org#https://mirrors.163.com#g' /etc/apt/sources.list
#2 CACHED

#3 [internal] load build definition from Dockerfile
#3 transferring dockerfile: 1.09kB done
#3 DONE 0.0s

#4 [internal] load .dockerignore
#4 transferring context: 2B done
#4 DONE 0.0s

#5 [internal] load metadata for docker.io/getsentry/snuba:22.1.0
#5 DONE 0.0s

#6 [internal] load build context
#6 transferring context: 95B done
#6 DONE 0.0s

#7 [3/6] RUN sed -i 's|#*deb http://security.debian.org|deb https://mirrors.cqu.edu.cn|' /etc/apt/sources.list /etc/apt/sources.list
#7 DONE 0.7s

#8 [4/6] RUN  apt-get clean && chmod +x /usr/sbin/policy-rc.d && echo "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d
#7 DONE 0.7s
#7 DONE 0.7s

#8 [4/6] RUN  apt-get clean && chmod +x /usr/sbin/policy-rc.d && echo "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d
#7 DONE 0.8s

#8 [4/6] RUN  apt-get clean && chmod +x /usr/sbin/policy-rc.d && echo "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d

#8 [4/6] RUN  apt-get clean && chmod +x /usr/sbin/policy-rc.d && echo "#!/bin/sh\nexit 0" > /usr/sbin/policy-rc.d
#8 DONE 0.7s
#8 DONE 0.7s

#9 [5/6] RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*

#9 [5/6] RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
#8 DONE 0.8s

#9 [5/6] RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
#8 DONE 0.8s

#9 [5/6] RUN apt-get update && apt-get install -y --no-install-recommends cron &&     rm -r /var/lib/apt/lists/*
#9 1.035 Get:1 https://mirrors.cqu.edu.cn/debian-security bullseye-security InRelease [48.4 kB]
#9 1.035 Get:1 https://mirrors.cqu.edu.cn/debian-security bullseye-security InRelease [48.4 kB]
#9 1.300 Get:2 https://mirrors.163.com/debian bullseye InRelease [116 kB]
#9 1.300 Get:2 https://mirrors.163.com/debian bullseye InRelease [116 kB]
#9 1.147 Get:1 https://mirrors.cqu.edu.cn/debian-security buster/updates InRelease [34.8 kB]
#9 1.423 Get:3 https://mirrors.cqu.edu.cn/debian-security bullseye-security/main amd64 Packages [239 kB]
#9 1.423 Get:3 https://mirrors.cqu.edu.cn/debian-security bullseye-security/main amd64 Packages [239 kB]
#9 1.418 Get:2 https://mirrors.163.com/debian buster InRelease [122 kB]
#9 1.528 Get:3 https://mirrors.cqu.edu.cn/debian-security buster/updates/main amd64 Packages [486 kB]
#9 1.870 Get:4 https://mirrors.163.com/debian bullseye-updates InRelease [44.1 kB]
#9 1.870 Get:4 https://mirrors.163.com/debian bullseye-updates InRelease [44.1 kB]
#9 2.071 Get:5 https://mirrors.163.com/debian bullseye/main amd64 Packages [8183 kB]
#9 2.071 Get:5 https://mirrors.163.com/debian bullseye/main amd64 Packages [8183 kB]
#9 2.543 Get:4 https://mirrors.163.com/debian buster-updates InRelease [56.6 kB]
#9 4.013 Get:5 https://mirrors.163.com/debian buster/main amd64 Packages [7909 kB]
#9 106.9 Get:6 https://mirrors.163.com/debian buster-updates/main amd64 Packages [8788 B]
#9 108.9 Fetched 8618 kB in 1min 48s (79.6 kB/s)
#9 108.9 Reading package lists...
#9 110.1 Reading package lists...
#9 111.5 Building dependency tree...
#9 111.8 Reading state information...
#9 111.9 The following additional packages will be installed:
#9 111.9   lsb-base sensible-utils
#9 111.9 Suggested packages:
#9 111.9   anacron logrotate checksecurity
#9 111.9 Recommended packages:
#9 111.9   default-mta | mail-transport-agent
#9 112.0 The following NEW packages will be installed:
#9 112.0   cron lsb-base sensible-utils
#9 112.8 0 upgraded, 3 newly installed, 0 to remove and 46 not upgraded.
#9 112.8 Need to get 143 kB of archives.
#9 112.8 After this operation, 383 kB of additional disk space will be used.
#9 112.8 Get:1 https://mirrors.163.com/debian buster/main amd64 sensible-utils all 0.0.12 [15.8 kB]
#9 113.2 Get:2 https://mirrors.163.com/debian buster/main amd64 lsb-base all 10.2019051400 [28.4 kB]
#9 113.4 Get:6 https://mirrors.163.com/debian bullseye-updates/main amd64 Packages [14.6 kB]
#9 113.4 Get:6 https://mirrors.163.com/debian bullseye-updates/main amd64 Packages [14.6 kB]
#9 113.6 Get:3 https://mirrors.163.com/debian buster/main amd64 cron amd64 3.0pl1-134+deb10u1 [99.0 kB]
#9 114.5 debconf: delaying package configuration, since apt-utils is not installed
#9 114.5 Fetched 143 kB in 2s (65.9 kB/s)
#9 114.6 Selecting previously unselected package sensible-utils.
(Reading database ... 11935 files and directories currently installed.)
#9 114.6 Preparing to unpack .../sensible-utils_0.0.12_all.deb ...
#9 114.6 Unpacking sensible-utils (0.0.12) ...
#9 114.6 Selecting previously unselected package lsb-base.
#9 114.6 Preparing to unpack .../lsb-base_10.2019051400_all.deb ...
#9 114.6 Unpacking lsb-base (10.2019051400) ...
#9 114.7 Selecting previously unselected package cron.
#9 114.7 Preparing to unpack .../cron_3.0pl1-134+deb10u1_amd64.deb ...
#9 114.7 Unpacking cron (3.0pl1-134+deb10u1) ...
#9 114.8 Setting up lsb-base (10.2019051400) ...
#9 114.8 Setting up sensible-utils (0.0.12) ...
#9 114.8 Setting up cron (3.0pl1-134+deb10u1) ...
#9 114.9 Adding group `crontab' (GID 101) ...
#9 114.9 Done.
#9 114.9 invoke-rc.d: could not determine current runlevel
#9 115.2 Fetched 8644 kB in 1min 55s (75.5 kB/s)
#9 115.2 Reading package lists...#9 115.2 Fetched 8644 kB in 1min 55s (75.5 kB/s)
#9 115.2 Reading package lists...#9 115.0 Starting periodic command scheduler: cron.
#9 DONE 115.4s

#10 [6/6] COPY entrypoint.sh /entrypoint.sh
#10 DONE 0.0s

#11 exporting to image
#11 exporting layers 0.1s done
#11 writing image sha256:d72d1c630b93a8618dc95448f2b0f056a3b8a4e58a534602805478804f820b6a done
#11 naming to docker.io/library/sentry-cleanup-self-hosted-local done
#11 DONE 0.1s

#9 116.5 Reading package lists...#9 116.5 Reading package lists...
#9 117.5 Building dependency tree...
#9 117.5 Building dependency tree...
#9 117.7 Reading state information...

#9 117.7 Reading state information...
#9 117.9 The following additional packages will be installed:
#9 117.9   sensible-utils
#9 117.9 Suggested packages:
#9 117.9   anacron logrotate checksecurity
#9 117.9 Recommended packages:
#9 117.9   default-mta | mail-transport-agent
#9 117.9 The following NEW packages will be installed:#9 117.9 The following additional packages will be installed:
#9 117.9   sensible-utils
#9 117.9 Suggested packages:
#9 117.9   anacron logrotate checksecurity
#9 117.9 Recommended packages:
#9 117.9   default-mta | mail-transport-agent
#9 117.9 The following NEW packages will be installed:

#9 117.9   cron sensible-utils
#9 117.9   cron sensible-utils
#9 123.6 0 upgraded, 2 newly installed, 0 to remove and 38 not upgraded.
#9 123.6 Need to get 114 kB of archives.
#9 123.6 After this operation, 314 kB of additional disk space will be used.
#9 123.6 0 upgraded, 2 newly installed, 0 to remove and 38 not upgraded.
#9 123.6 Need to get 114 kB of archives.
#9 123.6 After this operation, 314 kB of additional disk space will be used.
#9 123.6 Get:1 https://mirrors.163.com/debian bullseye/main amd64 sensible-utils all 0.0.14 [14.8 kB]
#9 123.6 Get:1 https://mirrors.163.com/debian bullseye/main amd64 sensible-utils all 0.0.14 [14.8 kB]
#9 124.1 Get:2 https://mirrors.163.com/debian bullseye/main amd64 cron amd64 3.0pl1-137 [99.6 kB]
#9 124.1 Get:2 https://mirrors.163.com/debian bullseye/main amd64 cron amd64 3.0pl1-137 [99.6 kB]
#9 125.0 debconf: delaying package configuration, since apt-utils is not installed
#9 125.0 debconf: delaying package configuration, since apt-utils is not installed
#9 125.0 Fetched 114 kB in 7s (16.8 kB/s)
#9 125.1 Selecting previously unselected package sensible-utils.
#9 125.0 Fetched 114 kB in 7s (16.8 kB/s)
#9 125.1 Selecting previously unselected package sensible-utils.
(Reading database ... 7023 files and directories currently installed.)
#9 125.1 Preparing to unpack .../sensible-utils_0.0.14_all.deb ...
#9 125.1 Unpacking sensible-utils (0.0.14) ...
#9 125.1 Selecting previously unselected package cron.
#9 125.1 Preparing to unpack .../cron_3.0pl1-137_amd64.deb ...
#9 125.1 Unpacking cron (3.0pl1-137) ...
#9 125.2 Setting up sensible-utils (0.0.14) ...
(Reading database ... 7023 files and directories currently installed.)
#9 125.1 Preparing to unpack .../sensible-utils_0.0.14_all.deb ...
#9 125.1 Unpacking sensible-utils (0.0.14) ...
#9 125.1 Selecting previously unselected package cron.
#9 125.1 Preparing to unpack .../cron_3.0pl1-137_amd64.deb ...
#9 125.1 Unpacking cron (3.0pl1-137) ...
#9 125.2 Setting up sensible-utils (0.0.14) ...
#9 125.2 Setting up cron (3.0pl1-137) ...
#9 125.3 Adding group `crontab' (GID 101) ...
#9 125.2 Setting up cron (3.0pl1-137) ...
#9 125.3 Adding group `crontab' (GID 101) ...
#9 125.3 Done.
#9 125.4 invoke-rc.d: could not determine current runlevel
#9 125.4 Starting periodic command scheduler: cron.
#9 125.3 Done.
#9 125.4 invoke-rc.d: could not determine current runlevel
#9 125.4 Starting periodic command scheduler: cron.
#9 DONE 125.8s

#10 [6/6] COPY entrypoint.sh /entrypoint.sh
#10 DONE 0.0s

#11 exporting to image
#11 exporting layers
#9 DONE 125.8s

#10 exporting to image
#10 exporting layers
#11 exporting layers 0.1s done
#11 writing image sha256:d7d883a2e0a487ad040ad5b46b5dca063331c8f3925e3b35dbdc4db6946bded3 done
#11 naming to docker.io/library/snuba-cleanup-self-hosted-local done
#11 DONE 0.1s
#10 exporting layers 0.1s done
#10 writing image sha256:a649821b710d799f6740fa4cbdee6a96445972925bbf47def0425deb79610525 done
#10 naming to docker.io/library/snuba-cleanup-self-hosted-local done
#10 DONE 0.1s
#9 CANCELED

This step has been stuck and didn't proceed to the next step for 4-5 hours.

**Comment by hubertdeng123** [comment]
Interesting, have you tried restarting the install script to see if it gets stuck there every time? Is this a clean installation?

**Comment by enjoy2303** [comment]
> Interesting, have you tried restarting the install script to see if it gets stuck there every time? Is this a clean installation?

It's a fresh installation, and I've tried several times. Every time it gets stuck at this point. I checked the installation logs and found the following error messages: "#9 125.4 invoke-rc.d: could not determine current runlevel #9 125.4 Starting periodic command scheduler: cron." I'm wondering if these two error messages may be related to the issue, but I don't know how to troubleshoot it.
